### ISSUE TITLE
Slack backend send_message to channel

### ISSUE THREAD

**Issue by ethan-eb** [errbotio/errbot#767]
Running errbot 4.1.3 under python3 (related issue #713 was under python2.7) in a virtualenv on Ubuntu 16.04.

My plugin is polling with `start_poller` callbacks. The callbacks retrieve a list of channels `["#foo", "#bar"]` and users `["@bob", "@roy"]` and then pass along messages as so:

```python
        for u in user_list:
            self.send(self.build_identifier(u), msg)
        for c in channels_list:
            self.send(self.build_identifier(c), msg)
```

This works fine for the users, but when attempting to send to a channel:

    Traceback (most recent call last):
      File "/usr/local/www/evbot/releases/current/venv/lib/python3.5/site-packages/errbot/backends/slack.py", line 487, in send_message
        to_channel_id = self.get_im_channel(self.username_to_userid(mess.to.username))
      File "/usr/local/www/evbot/releases/current/venv/lib/python3.5/site-packages/errbot/backends/slack.py", line 406, in username_to_userid
        raise UserDoesNotExistError("Cannot find user %s" % name)
    errbot.backends.base.UserDoesNotExistError: Cannot find user <None>

Could it be a problem in `backends/slack.py` *build_identifier*? There's no case that returns `SlackRoom` only `SlackRoomOccupant` ... is that correct in the case where the `txtrep` is "#foo"?

**Comment by zoni** [comment]
I think this is related to #761. It's on my radar but I have a lot going on so I haven't had time to join in on the issue/discussion yet unfortunately.

**Comment by gbin** [comment]
This is #778 we also saw it while reviewing #761.

**Comment by gbin** [comment]
It should be fixed by 0e7652da523e6abdede9657e0883da476d8ead03

Please test and reopen if you detect a problem.

**Comment by ethan-eb** [comment]
That fixed our problem, thanks!
